icu4c/source/i18n/usearch.cpp

```diff
@@ -545,2048 +541,79 @@ inline void setShiftTable(int16_t   shift[], int16_t backshift[],
-
-    if (textoffset &&
-        ucol_unsafeCP(strsrch->canonicalPrefixAccents[
-                                 u_strlen(strsrch->canonicalPrefixAccents) - 1
-                                         ], collator)) {
-        safeoffset     = getNextSafeOffset(collator, text, textoffset,
-                                           strsrch->search->textLength);
-        safelength     = safeoffset - textoffset;
-        safetextlength = INITIAL_ARRAY_SIZE_;
-        safetext       = addToUCharArray(safebuffer, &safetextlength,
-                                         strsrch->canonicalPrefixAccents,
-                                         text + textoffset, safelength,
-                                         NULL, status);
-    }
-    else {
-        safetextlength = u_strlen(strsrch->canonicalPrefixAccents);
-        safetext       = strsrch->canonicalPrefixAccents;
-    }
-
-    UCollationElements *coleiter = strsrch->utilIter;
-     // if status is a failure, ucol_setText does nothing
-    ucol_setText(coleiter, safetext, safetextlength, status);
-    // status checked in loop below
-
-    int32_t  *ce           = strsrch->pattern.ces;
-    int32_t   celength     = strsrch->pattern.cesLength;
-    int       ceindex      = 0;
-    UBool     isSafe       = TRUE; // safe zone indication flag for position
-    int32_t   prefixlength = u_strlen(strsrch->canonicalPrefixAccents);
-
-    while (ceindex < celength) {
-        int32_t textce = ucol_next(coleiter, status);
-        if (U_FAILURE(*status)) {
-            if (isSafe) {
-                cleanUpSafeText(strsrch, safetext, safebuffer);
-            }
-            return USEARCH_DONE;
-        }
-        if (textce == UCOL_NULLORDER) {
-            // check if we have passed the safe buffer
-            if (coleiter == strsrch->textIter) {
-                cleanUpSafeText(strsrch, safetext, safebuffer);
-                return USEARCH_DONE;
-            }
-            cleanUpSafeText(strsrch, safetext, safebuffer);
-            safetext = safebuffer;
-            coleiter = strsrch->textIter;
-            setColEIterOffset(coleiter, safeoffset);
-            // status checked at the start of the loop
-            isSafe = FALSE;
-            continue;
-        }
-        textce = getCE(strsrch, textce);
-        if (textce != UCOL_IGNORABLE && textce != ce[ceindex]) {
-            // do the beginning stuff
-            int32_t failedoffset = ucol_getOffset(coleiter);
-            if (isSafe && failedoffset <= prefixlength) {
-                // alas... no hope. failed at rearranged accent set
-                cleanUpSafeText(strsrch, safetext, safebuffer);
-                return USEARCH_DONE;
-            }
-            else {
-                if (isSafe) {
-                    failedoffset = safeoffset - failedoffset;
-                    cleanUpSafeText(strsrch, safetext, safebuffer);
-                }
+inline void initialize(UStringSearch *strsrch, UErrorCode *status)
```

Comment:
The comment for this `initialize()` function says "Building of the pattern collation element list and the boyer moore strsrch table."

Also, the preceding function, `setShiftTable()`, looks like it's setting up Boyer-Moore stuff.

But, in the interest of keeping this PR simple, quick and mechanical, Ietting these slide seems OK.

Reply:
Good point, thanks Andy!

I noticed a few other places in this file where things could be cleaned up / changed as well.
I think I'll create another PR after this one to make some additional changes (using the same ticket number, since they are related to the Boyer-Moore code). That way this PR is just a simple removal, without other changes mixed in with it.